### PR TITLE
[terra-form-Input][terra-form-InputField] Added placeholder to Input and Input Field.

### PR DESCRIPTION
**Pull request by ST063655**
### Summary

**What was changed:**
Added Placeholder to Input and Input Field.

**Why it was changed:**
For Mpages Terra Passthrough, Placeholder was added to Input and Input Field.

### Testing

This change was tested using:

- [ ] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9153 

**Placeholder Example for Terra-form-InputField:**

Screenshot 2023-06-08 at 9 07 26 PM

**Placeholder Example for Terra-form-Input:**

Screenshot 2023-06-08 at 7 48 49 PM

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-core-docs/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+*Added
+  * Added placeholder for terra-form-input tests.
+  * Added placeholder for terra-form-inputfield tests.
```
Comment (supreethmr) [inline packages/terra-core-docs/CHANGELOG.md:7]:
```suggestion
  * Added placeholder for terra-form-input tests.
  * Added placeholder for terra-form-inputfield tests.
```
we can add this two entries under below Added section instead of creating new.

File: packages/terra-core-docs/src/terra-dev-site/doc/form-input/About.1.doc.mdx [hunk 2]
```diff
@@ -15,6 +15,7 @@ import UncontrolledDefaultInvalidExample from './example/uncontrolled/DefaultInv
 import UncontrolledDefaultIncompleteExample from './example/uncontrolled/DefaultIncompleteExample?dev-site-example';
 import UncontrolledInvalidWithIncompleteExample from './example/uncontrolled/InvalidWithIncompleteExample?dev-site-example';
 import UncontrolledDisabledInvalidExample from './example/uncontrolled/DisabledInvalidExample?dev-site-example';
+import PlaceholderExample from './example/uncontrolled/PlaceholderExample?dev-site-example';
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/doc/form-input/About.1.doc.mdx:18]: let's remove this examples from docs. As we don't want consumers to use this new MPage attribute to render placeholder.

File: packages/terra-core-docs/src/terra-dev-site/doc/form-input/InputField.2.doc.mdx [hunk 3]
```diff
@@ -4,6 +4,7 @@ import InputFieldExample from './example/InputField?dev-site-example';
 import RequiredInputFieldExample from './example/RequiredInputField?dev-site-example';
 import IncompleteInputFieldExample from './example/IncompleteInputField?dev-site-example';
 import NumberInputFieldExample from './example/NumberInputField?dev-site-example';
+import PlaceholderInputFieldExample from './example/PlaceholderInputField?dev-site-example';
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/doc/form-input/InputField.2.doc.mdx:7]: same here !! we need delete these examples as well

File: packages/terra-form-input/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* Added
+   * Added placeholder to Input and Input Field.
+   * Added placeholder examples for Input and Input Field.
```
Comment (supreethmr) [inline packages/terra-form-input/CHANGELOG.md:7]:
```suggestion
   * Enabled placeholder support for `fusion-textbox`.
```

File: packages/terra-form-input/tests/jest/Input.test.jsx [hunk 5]
```diff
@@ -17,6 +17,14 @@ describe('Input', () => {
     expect(wrapper).toMatchSnapshot();
   });
 
+  it('should render as uncontrolled when placeholder is passed into the Input', () => {
```
Comment (supreethmr) [inline packages/terra-form-input/tests/jest/Input.test.jsx:20]:
```suggestion
  it('should render placeholder when showMpageFusionPlaceholder is passed into the Input', () => {
```

File: packages/terra-form-input/tests/jest/__snapshots__/InputField.test.jsx.snap [hunk 6]
```diff
@@ -1,5 +1,38 @@
...
+>
+  <Input
+    disabled={false}
+    id="test-input"
+    isIncomplete={false}
+    isInvalid={false}
+    name={null}
+    placeholder="Enter The Text"
+    required={false}
+    showMpageFusionPlaceholder={true}
```
Comment (supreethmr) [inline packages/terra-form-input/tests/jest/__snapshots__/InputField.test.jsx.snap:31]: this should not be added to Input !!

Comment (ST063655) [reply packages/terra-form-input/tests/jest/__snapshots__/InputField.test.jsx.snap]: Removed the showMpageFusionPlaceholder={true}.

File: packages/terra-form-input/tests/jest/Input.test.jsx (hunk 5 above)
Comment (ST063655) [reply packages/terra-form-input/tests/jest/Input.test.jsx]: Commit suggestion.

File: packages/terra-core-docs/src/terra-dev-site/doc/form-input/InputField.2.doc.mdx (hunk 3 above)
Comment (ST063655) [reply packages/terra-core-docs/src/terra-dev-site/doc/form-input/InputField.2.doc.mdx]: Removed the example.

File: packages/terra-core-docs/src/terra-dev-site/doc/form-input/About.1.doc.mdx (hunk 2 above)
Comment (ST063655) [reply packages/terra-core-docs/src/terra-dev-site/doc/form-input/About.1.doc.mdx]: Removed the example.

File: packages/terra-core-docs/CHANGELOG.md (hunk 1 above)
Comment (ST063655) [reply packages/terra-core-docs/CHANGELOG.md]: Added the entries.

File: packages/terra-form-input/CHANGELOG.md (hunk 4 above)
Comment (ST063655) [reply packages/terra-form-input/CHANGELOG.md]: Added.

File: packages/terra-core-docs/CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,8 @@
 * Added
   * Added terra-scroll A11y tests.
   * Added an example in terra-content-container without interactive elements.
+  * Added placeholder for terra-form-input tests.
+  * Added placeholder for terra-form-inputfield tests.
```
Comment (supreethmr) [inline packages/terra-core-docs/CHANGELOG.md:9]: Do you still need them ??

File: packages/terra-form-input/tests/jest/InputField.test.jsx [hunk 8]
```diff
@@ -21,6 +21,14 @@ describe('InputField', () => {
     expect(wrapper).toMatchSnapshot();
   });
 
+  it('should render placeholder when showMpageFusionPlaceholder is passed', () => {
+    const textarea = <InputField inputId="test-input" label="label" inputAttrs={{ placeholder: 'Enter The Text' }} />;
+    const wrapper = shallow(textarea);
+    const placeholderInput = (wrapper.find('Field')).find('Input#test-input');
+    expect(placeholderInput.props()).toHaveProperty('placeholder', 'Enter The Text');
```
Comment (supreethmr) [inline packages/terra-form-input/tests/jest/InputField.test.jsx:28]: as per the code `<InputField inputId="test-input" label="label" inputAttrs={{ placeholder: 'Enter The Text' }} />` placeholder should be null here, since `showMpageFusionPlaceholder` prop is not passed placeholder value should be null in this case

File: packages/terra-form-input/tests/jest/Input.test.jsx [hunk 9]
```diff
@@ -17,6 +17,14 @@ describe('Input', () => {
     expect(wrapper).toMatchSnapshot();
   });
 
+  it('should render placeholder when showMpageFusionPlaceholder is passed into the Input', () => {
+    const input = <Input defaultValue="foo" id="placeholderText" ariaLabel="label" placeholder="Enter The Text" showMpageFusionPlaceholder />;
+    const wrapper = shallow(input);
+    const placeholderInput = wrapper.find('input#placeholderText');
+    expect(placeholderInput.props()).toHaveProperty('placeholder', 'Enter The Text');
```
Comment (supreethmr) [inline packages/terra-form-input/tests/jest/Input.test.jsx:24]: we need to test the other scenario as well here ensure  placeholder is set to null when you pass placeholder without showMpageFusionPlaceholder prop

File: packages/terra-form-input/tests/jest/Input.test.jsx [hunk 10]
```diff
@@ -17,6 +17,22 @@ describe('Input', () => {
...
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should render placeholder when showMpageFusionPlaceholder is not passed into the Input', () => {
```
Comment (supreethmr) [inline packages/terra-form-input/tests/jest/Input.test.jsx:28]:
```suggestion
  it('should placeholder with null when showMpageFusionPlaceholder is not passed into the Input', () => {
```

**Review by saket2403 — APPROVED** [review]

**Review by supreethmr — APPROVED** [review]
